### DISCUSSION TITLE
Suggestions for runtime editing tools?

### DISCUSSION THREAD

**Discussion by jasonswearingen** [spimort/TerraBrush#122] [General]
Hello, I was wondering what suggestions for runtime editing tools/gizmos/plugins are around for C# devs?  (gdscript suggestions are okay, as I can use those for reference purposes)

Specifically, I want to allow players to make terrain/building/prop changes in-game, so I need a ui layer to make this work nicely.  I am planning on making this myself, but if there are some tips on where I can find similar features, I won't have to recreate the wheel.

Thank you

FYI, I found this, which looks good for placement snapping:  https://github.com/mharitsnf/ExtraSnaps/tree/master

**Comment by spimort** [comment]
hey @jasonswearingen, that's a great idea! 🙂

I don't know much about ingame gizmos, but the only that comes to my mind is this game : 
https://store.steampowered.com/app/1043390/FlowScape/

It's not a Godot game (I think), but it could be a good point of reference!

It would totally be possible to do a UI on top of TerraBrush for ingame editing.

Another thing that comes to my mind is asset placer. It's not really an InGame thing but still, maybe for reference?
https://cookiebadger.itch.io/assetplacer

In theory, you could reuse the pie menu from TerraBrush, it's not really attached to anything specific to TerraBrush.

I hope this helps, let me know!
